src/main/java/seedu/address/model/ItemModelManager.java

```diff
@@ -463,11 +469,8 @@ public boolean togglePriorityMode() throws IllegalListException {
 
     private VisualizeList getNextTask() {
         TaskList result = new TaskList();
-        if (sortedTask.peek().getTask().get().isComplete()) {
-            sortedTask.poll();
-        }
 
-        if (sortedTask.size() == 0) {
+        if (sortedTask.peek().getTask().get().isComplete()) {
```

Comment:
It seems like you repeat the method chaining .getTask().get().isComplete() quite often. Perhaps you can create a method in Item that checks if the task is complete? We just learnt the Law of Demeter, let's put it to good use :D 
Also, as sortedTask is a collection, perhaps a plural name would be more suitable?